cmake/CMakeLists.txt

```diff
@@ -402,10 +402,26 @@ if (WIN32)
       endforeach()
     endif()
 else()
+  # Enabling this for Mac builds causes a sigabrt in several tests. Secondly the right option for Mac is -dead_strip
+  if (UNIX AND NOT APPLE)
+    foreach(type EXE SHARED)
+      set(CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO} -Wl,--gc-sections")
+      set(CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO} -Wl,--gc-sections")
+      set(CMAKE_${type}_LINKER_FLAGS_RELEASE "${CMAKE_${type}_LINKER_FLAGS_RELEASE} -Wl,--gc-sections")
+      set(CMAKE_${type}_LINKER_FLAGS_RELEASE "${CMAKE_${type}_LINKER_FLAGS_RELEASE} -Wl,--gc-sections")
```

Comment:
Could you please only apply these flags to onnxruntime.dll and the python binding? That will have less impact to the whole project.

Reply:
doesn't ```foreach(type EXE SHARED)``` ensure that?

Reply:
If we apply them to all the binaries, it may cause unexpected side effect.  At least, protoc do not need this flag.

Reply:
Ok, let me check. I was just following the Windows example right above this.

Reply:
done.